### PR TITLE
Docs: add a Frequently Asked Questions page

### LINKED ISSUE #3965: Create a FAQ for the documentation
This should contain commonly asked questions relating to problems that arise after people upgrade their installation etc. There should be a link on the README.md and maybe we should add an issue template that tells people to first consult the FAQ before submitting an issue.

### PR DESCRIPTION
**Pull request by sphuber**
Fixes #3965 

Currently, the new FAQ page mostly focuses on common problems that
users encounter and direct either towards the mailing list or create
issues for. This FAQ page should become the central place to redirect to
in case of these common problems.

Besides the FAQ page itself, there are some other minor changes:

 * Improve `update installation` instructions to include the necessity
   of running `reentry` scan, and to be even more clear that the daemon
   should absolutely be stopped before upgrading, after having finished
   all running processes first.
 * Include link to FAQ page in the README
 * Remove outdated section from the Troublehooting section, since it was
   about Postgres versions that are no longer supported.

### CODE REVIEW

**Comment by sphuber** [comment]
Currently, the FAQ entry in the sidebar can be expanded where each individual question becomes an item, because I use section markup for the questions. I am personally not sold on this, but am not sure how to suppress this. If you agree that it is a bit messy to have those long titles in the side bar and you know how to get rid of them, please let me know.

**Comment by sphuber** [comment]
@csadorf if we find a place in the scaffolding for the FAQ, we could already review this and merge it I think. This is a new addition that I created before the docs reorganization.

**Comment by csadorf** [comment]
> @csadorf if we find a place in the scaffolding for the FAQ, we could already review this and merge it I think. This is a new addition that I created before the docs reorganization.

Some of the questions addressed here should be moved to the "Troubleshooting" section as part of the intro chapter. The remaining content could then be placed as a distinct top-level section right after the How-Tos with the assumption that users should focus on going through the tutorial and relevant How-Tos before it makes sense to them to turn to FAQs. We then replace the "Development" panel with a FAQ panel on the landing page to "advertise" this more prominently. Thoughts?

**Comment by ltalirz** [comment]
> The remaining content could then be placed as a distinct top-level section

This is one possibility but I find it a bit unflexible as there can be frequently asked questions on any stage - installation, tutorial, managing data, ...

E.g. I also still have on my to-do list to convert a document into a Q&A format that we prepared for the delegation of the Japanese National Institute of Informatics, which would be aimed more at the type of questions a manager would ask in order to decide whether AiiDA is worth giving a try (I'll send it to you in private).

**Comment by csadorf** [comment]
As decided during team meeting we will place the FAQ section as the last subsection within the HowTo chapter for now.

**Comment by csadorf** [comment]
@sphuber Can you rebase this on docs-revamp?

**Comment by sphuber** [comment]
@csadorf rebased and ready for review

**Comment by csadorf** [comment]
@sphuber I am very sorry, but looks like another rebase may be necessary.

**Comment by sphuber** [comment]
not to worry. Rebase coming right up

**Comment by sphuber** [comment]
@csadorf batter up!

File: docs/source/howto/faq.rst [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+However, be careful, if you make these intervals too short, the daemon workers may spam the remote machine and/or scheduler, which could have adverse effects on the machine itself or can get your account banned, depending on the policy of the remote machine.
+An additional note of importance is that each interval is guaranteed to be respected per daemon worker individually, but not as a collective.
+That is to say, if the safe interval is set to 60 seconds, any single worker is guaranteed to open a connection to that machine at most once every minute, however, if you have multiple active daemon workers, the machine may be accessed more than once per minute.
+
+
+When I run a process, it finishes without any problems, but if I submit it to the daemon, it excepts. What is happening?
+========================================================================================================================
+In 99 out of a 100 cases, this is because a piece of code that is being executed by the process is not importable by the daemon.
+To determine exactly what, first set the loglevel to ``DEBUG`` by executing the command ``verdi config logging.aiida_loglevel DEBUG``.
```
Comment (csadorf) [inline docs/source/howto/faq.rst:58]:
```suggestion
This is almost always caused by an import issue.
To determine exactly what might be going wrong, first set the loglevel to ``DEBUG`` by executing the command ``verdi config logging.aiida_loglevel DEBUG``.
```

File: docs/source/howto/faq.rst [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+Then restart the daemon with ``verdi daemon restart`` for the changes to tak effect.
```
Comment (csadorf) [inline docs/source/howto/faq.rst:59]:
```suggestion
Then restart the daemon with ``verdi daemon restart`` for the changes to take effect.
```

File: docs/source/howto/faq.rst [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+Run the command ``verdi daemon logshow`` in a separate terminal to see the logging output of the daemon and then submit the problematic calculation or workflow once more.
```
Comment (csadorf) [inline docs/source/howto/faq.rst:60]:
```suggestion
Run the command ``verdi daemon logshow`` in a separate terminal to see the logging output of the daemon and then submit the problematic calculation or workflow again.
```

File: docs/source/howto/faq.rst [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+I have updated the version of AiiDA and now it is no longer working. What should I do?
+======================================================================================
+First, make sure that your daemon is not running.
+You can check this with ``verdi daemon status``.
+If you find that your daemon was actually still running, that is likely the problem, so stop it first using ``verdi daemon stop``.
+It is very important that each time you want to update your AiiDA installation, you should *always* first finish all running processes and stop the daemon before doing so.
+After you have stopped the daemon, make sure to run ``reentry scan`` before you restart the daemon with ``verdi daemon start``.
+
+
+I get an MissingEntryPointError or MultipleEntryPoint exception, saying that a particular entry point cannot be found. How can I fix this?
```
[the remark this answers is not in the document]
Comment (sphuber) [reply docs/source/howto/faq.rst]: Turns out it is possible to have links in headers

**Comment by sphuber** [comment]
Thanks @csadorf . Addressed all the changes and added one additional change: for three of the "longer" commands, I used the appropriate code blocks. I have some of the more basic ones, like `verdi status` and `verdi daemon stop` as inline. Didn't feel it was necessary to have code blocks for those, but let me know if you disagree

File: docs/source/howto/faq.rst [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+Each time that you change your code, you should restart the daemon for the changes to take effect.
+Each daemon worker actually is its own system process with its own Python interpreter instance, and since we don't support automated hot==swapping, the daemon workers will not automatically detect the changes in the code.
+Simply calling ``verdi daemon restart`` will do the trick.
+
+
+I have updated the code of a :py:class:`~aiida.engine.processes.workchains.WorkChain`, :py:class:`~aiida.engine.processes.calcjobs.CalcJob` or whatever other Python code, but my Python shell instance does not seem to pick up the changes?
+=============================================================================================================================================================================================================================================
+The ``verdi shell`` is its own Python interpreter and also not automatically detect code changes and reload them.
```
Comment (csadorf) [inline docs/source/howto/faq.rst:36]:
```suggestion
The ``verdi shell`` is its own Python interpreter and does not automatically detect code changes.
Restart the shell to reload the latest version of the code.
```

File: docs/source/howto/faq.rst [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+.. code-block:: console
+
+    $ verdi computer configure <TRANSPORT_TYPE> <COMPUTER> --safe-interval=<NUMBER_OF_SECONDS>
+
+and
+
+.. code-block:: ipython
+
+    In [1]:: computer.set_minimum_job_poll_interval(NUMBER_OF_SECONDS)
```
Comment (csadorf) [inline docs/source/howto/faq.rst:58]:
```suggestion
    In [1]: computer.set_minimum_job_poll_interval(NUMBER_OF_SECONDS)
```

File: docs/source/howto/installation.rst [hunk 7]
```diff
@@ -258,9 +258,12 @@ Here are a few general tips that might improve the AiiDA performance:
 Updating your installation
 ==========================
 
+Whenever updating your AiiDA installation, make sure you follow these instructions **very carefully**, even when merely upgrading the patch version!
+Failing to do so, may leave your installation in a broken state, or worse may even damage your data, potentially irreperably.
```
Comment (csadorf) [inline docs/source/howto/installation.rst:262]:
```suggestion
Failing to do so, may leave your installation in a broken state, or worse may even damage your data, potentially irreparably.
```

**Comment by sphuber** [comment]
> Just a few minor fixes, but otherwise this is good to go.

@csadorf done

**Comment by csadorf** [comment]
@ltalirz Please let me know if you would like to have another look at this, otherwise I'll merge, thx!

**Pull request merged**